lib/rufo/formatter.rb

```diff
@@ -2673,6 +2759,55 @@ def visit_literal_elements(elements, inside_hash: false, inside_array: false, to
     end
   end
 
+  def add_comments_to_doc(comments, doc)
+    return false if comments.empty?
+
+    comments.each do |c|
+      doc << B.line_suffix(" " + c.rstrip)
+    end
+    return true
+  end
+
+  def visit_literal_elements_doc(elements, inside_hash: false, inside_array: false, token_column: false)
+    doc = []
+    pre_comments = []
+    post_comments = []
+
+    comments, newline_before_comment = skip_space_or_newline_doc
+    has_comment = add_comments_to_doc(comments, pre_comments)
+
+    elements.each_with_index do |elem, i|
+      doc_el = visit(elem)
+      if doc_el.is_a?(Array)
```

Comment:
Nitpick: what do you think about coalescing every `doc_el` into an array? This can replace this entire `if` expression

```ruby
doc.concat Array(doc_el)
```

Reply:
I have tried this and it causes the tests to break. Perhaps we can revisit this once we are further down the implementation of prettier printer.